src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -466,6 +466,17 @@ public Object eval (DataInstance model, EvaluationContext evalContext) {
                 (String) argVals[0],
                 args.length == 3 ? Encoding.from((String)argVals[2]) : Encoding.BASE64
             );
+        } else if (name.equals("randomize")) {
```

Comment:
I ran all tests with coverage, and we don’t seem to get here. Should we?

![screen shot 2018-04-25 at 6 57 49 pm](https://user-images.githubusercontent.com/55099/39281671-ba119160-48ba-11e8-93bb-f4d540a768bb.png)

Reply:
I suppose we should. The outside-in tests I have written use `randomize()` in a nodeset definition, which isn't eval'ed on this file. The only problem is that I can't think of a use for 'randomize()` out of that context.

If we can think of an example where `randomize()` could have sense outside a nodeset definition, I can write a new test. Any ideas, @lognaturel, @dcbriccetti, @yanokwa?

Reply:
Or maybe we delete this code block, 469–479?
I’m looking for how to hook in with the debugger and watch this new code run when processing, for example, this:
`<itemset nodeset="randomize(instance('fruit')/root/item)">`

Reply:
That code won't touch that block because we don't actually eval the nodeset. We would need to come up with some other use for `randomize()` that is not inside an itemset nodeset definition. 

Let's give it some days. Maybe @lognaturel or @yanokwa can come up with some reasonable use for `randomize()` and I can write a test that uses that block.

Reply:
I can't think of a super meaningful example and that's why this narrow support isn't so bad.

But just for testing, you can do something silly in a calculate like `max(randomize(/path/to/int/in/repeat))`. You can use any of the [functions](https://opendatakit.github.io/xforms-spec/#xpath-functions) that take a nodeset parameter. Order doesn't matter for any of them so they won't be super meaningful but they'll exercise the code.

Reply:
Just to make it a little less silly, how could we take the first element of the output of `randomize(/path/to/int/in/repeat)`?

Reply:
Good idea! I think you can add `[1]` or `[position() = 1]`. That is actually useful. 👍